app/src/main/java/com/github/polybooks/core/database/implementation/SaleDatabase.kt

```diff
@@ -72,62 +71,67 @@ class SaleDatabase : SaleDatabase {
         }
 
         override fun withOrdering(ordering: SaleOrdering): SaleQuery {
-            TODO("Not yet implemented")
+            this.ordering = ordering
+            return this
         }
 
-        override fun searchByISBN(isbn13: String): SaleQuery {
-            this.isbn13 = isbn13
+        override fun searchByISBN(isbn: ISBN): SaleQuery {
+            this.isbn = isbn
             return this
         }
 
-        private fun getQuery() : Query {
-            var query: Query = saleRef
+        private fun filterQuery(q: Query) : Query {
+            var query: Query = q
 
-            // TODO: add these when necessary
-            // isbn13?.let { query = query.whereEqualTo("isbn", isbn13) }
-            // TODO: fix this for title
-            // FieldPath.of(SaleFields.BOOK.fieldName, BookFields.TITLE.fieldName)
-            title?.let { query = query.whereEqualTo(SaleFields.BOOK.fieldName + "." + BookFields.TITLE.fieldName, title) }//SaleFields.BOOK.fieldName[BookFields.TITLE.fieldName]
-            // interests?.let { query = query.whereIn("interests", interests!!.toList()) }
             states?.let { query = query.whereIn(SaleFields.STATE.fieldName, states!!.toList()) }
-            // TODO: fix this with whereIN
```

Comment:
you deleted this TODO, but is it fixed? Can we make query by both SaleState and BookCondition now?

Reply:
I thought it was fixed with the indices. I now added a test to check if it worked and it didn't. But I fixed it. It's ugly, but I think we don't have a choice.